homeassistant/components/generic/config_flow.py

```diff
@@ -261,15 +268,37 @@ async def async_test_stream(
     return {}
 
 
+def register_preview(hass: HomeAssistant, flow_id: str, user_input: dict[str, Any]):
+    """Set up previews for camera feeds during config flow."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if not hass.data[DOMAIN].get(PREVIEWS):
+        _LOGGER.debug("Registering camera image preview handler")
+        hass.http.register_view(CameraImagePreview(hass))
+    hass.data[DOMAIN][PREVIEWS] = {}
+    hass.data[DOMAIN][PREVIEWS][flow_id] = {}
+    hass.data[DOMAIN][PREVIEWS][flow_id]["user_input"] = user_input
```

Comment:
This feels hacky. I don't have a better suggestion at the moment.

Reply:
This was why i was holding off on merging for a second opinion originally.

Generally, how might a config flow share data with a view? Storing temporary data based on the specific flow seems like the best idea we have here.  Perhaps this comment is about the specific data that is shared here, so maybe reducing its scope or storing an object more tightly scoped to the function needed for viewing could be considered less hacky?

Reply:
Can we try storing the needed data in the flow context instead? We can get the flow context by looking up the partial flow result of a flow in progress with `hass.config_entries.flow.async_get`.

Reply:
Yeah, like that -- we were discussing that direction here https://github.com/home-assistant/core/pull/71269#discussion_r916413629 but I couldn't see how to do it.  Maybe the bit I couldn't see how to do was how to store the data in the context so that it could be accessible from the view via the flow result.

Reply:
If the flow object can be retrieved from the HTTP view context using just the flow ID then it could be done this way (and I agree would be better).

The problem is that the incoming GET requests don't have any other way of being linked back to the flow object.

Reply:
The flow context is a dict of type `dict[str, Any]`. Just set the data in an item in the context dict, instead of storing it in hass.data.

Example:
https://github.com/home-assistant/core/blob/21693b0022be231ca563906d90490310edec1f7f/homeassistant/components/apple_tv/config_flow.py#L161

Reply:
Sure, we can set data within the config_flow object.

But how will the GET request be able to find this, when it runs asynchronously and outside the scope of the config_flow?

Reply:
`hass.config_entries.flow.async_get` returns a partial `FlowResult` with the context as an attribute.

https://github.com/home-assistant/core/blob/d5b966d942a07036451505a4236d61d2884c7b68/homeassistant/data_entry_flow.py#L180-L185

https://github.com/home-assistant/core/blob/d5b966d942a07036451505a4236d61d2884c7b68/homeassistant/data_entry_flow.py#L109-L123

https://github.com/home-assistant/core/blob/d5b966d942a07036451505a4236d61d2884c7b68/homeassistant/data_entry_flow.py#L102

Use the `flow_id` in the endpoint to get a partial `FlowResult` and the get the context from that.

Reply:
When we last looked at this before I couldn't see where the context was set in the flow result, and i think `_async_flow_handler_to_flow_result` is what I was missing.

Reply:
@MartinHjelmare  thanks for the suggestion. Looks like it works 3214f41e!  Now the cam preview data is stored within the config flow.
As a result I simplified by removing all of the code associated with expiring old previews (on the assumption that old config flows are already expired and deleted).